File: labServiceUIJava/labServiceUIJava/src/main/java/com/example/labserviceuijava/service/ServiceOrder.java
package com.example.labserviceuijava.service;

import com.example.labserviceuijava.model.Order;
import com.example.labserviceuijava.repository.OrderDBRepository;
import com.example.labserviceuijava.repository.exceptions.RepositoryException;
import com.example.labserviceuijava.validator.OrderValidator;
import com.example.labserviceuijava.validator.ValidationException;

public class ServiceOrder {

    private final OrderDBRepository orderDBRepository;

    private final OrderValidator orderValidator;

    public ServiceOrder(OrderDBRepository orderDBRepository, OrderValidator orderValidator) {
        this.orderDBRepository = orderDBRepository;
        this.orderValidator = orderValidator;
    }

    public Order save(Integer concertID,String buyerName, Integer numberOfTickets){
        Order order=new Order(0,concertID,buyerName,numberOfTickets);
        try{
            orderValidator.validate(order);
        }catch(ValidationException validationException){
            System.out.println(validationException);
        }
        return orderDBRepository.save(order);
    }
}

File: labServiceUIcs/WindowsFormsApp2/service/ServiceOrder.cs
using System.Management.Instrumentation;
using WindowsFormsApp2.model;
using WindowsFormsApp2.repository;

namespace WindowsFormsApp2
{
    public class ServiceOrder
    {
        private OrdersDBRepository ordersDbRepository;

        public ServiceOrder(OrdersDBRepository ordersDbRepository)
        {
            this.ordersDbRepository = ordersDbRepository;
        }

        public Order save(int concertID, string buyerName, int numberOfTickets)
        {
            Order order = new Order(0, concertID, buyerName, numberOfTickets);
            return this.ordersDbRepository.Save(order);
        }
    }
}